browser/ui/webui/brave_education/education_page_ui.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/brave_education/education_page_ui.h"
+
+#include <string>
+#include <utility>
+
+#include "brave/browser/ui/webui/brave_education/education_page_delegate_desktop.h"
+#include "brave/browser/ui/webui/brave_education/education_page_handler.h"
+#include "brave/browser/ui/webui/brave_webui_source.h"
+#include "brave/components/brave_education/common/education_urls.h"
+#include "brave/components/brave_education/resources/grit/brave_education_generated_map.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/tabs/public/tab_interface.h"
+#include "chrome/browser/ui/webui/webui_util.h"
+#include "chrome/grit/branded_strings.h"
+#include "components/grit/brave_components_resources.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_ui.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "services/network/public/mojom/content_security_policy.mojom.h"
+
+namespace brave_education {
+
+EducationPageUI::EducationPageUI(content::WebUI* web_ui, const GURL& url)
```

Comment:
IIUC, since the URL is visible to the user, we would need to embed something like `chrome-untrusted://getting-started` inside of `chrome://getting-started`, where the `chrome-untrusted` frame would simply proxy the commands up to the outer UI. What would the benefits of that be?

It's worth noting that the supported commands found in `education_page.mojom` and implemented in `EducationPageHandler` have a strict set of requirements - they must be side-effect free.

@bridiver, do you have thoughts on this?

Reply:
> What would the benefits of that be?

It's a principle of the least privilege benefit. The page should only need access to the mojom APIs, not things like avoiding permissions prompting (e.g chrome:// doesn't need to user permission to access bluetooth API) and there's other more powerful capabilities granted to chrome:// that aren't granted to chrome-untrusted://. So, what we're doing is essentially restricting access to only the necessary features, not all available features to reduce the potential impact of an XSS or other vuln on the site page.